homeassistant/components/ws66i/__init__.py

```diff
@@ -0,0 +1,70 @@
+"""The Soundavo WS66i 6-Zone Amplifier integration."""
+import logging
+
+from pyws66i import get_ws66i
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_IP_ADDRESS
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .config_flow import CannotConnect, validate_input
+from .const import (
+    CONF_NOT_FIRST_RUN,
+    DOMAIN,
+    FIRST_RUN,
+    UNDO_UPDATE_LISTENER,
+    WS66I_OBJECT,
+)
+
+PLATFORMS = ["media_player"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Soundavo WS66i 6-Zone Amplifier from a config entry."""
+
+    # double negative to handle absence of value
+    first_run = not bool(entry.data.get(CONF_NOT_FIRST_RUN))
+    if first_run:
+        hass.config_entries.async_update_entry(
+            entry, data={**entry.data, CONF_NOT_FIRST_RUN: True}
+        )
+    else:
+        # Need to verify we can connect
+        try:
+            _ = await validate_input(hass, entry.data)
+        except CannotConnect as err:
+            raise ConfigEntryNotReady from err
```

Comment:
Check this in the config flow before creating the entry.  Don't create the entry if the input isn't valid

Reply:
I do validate user input in the config flow. I don't recall why I do it here in the async_setup_entry() method. I'm not sure if I added in response to some testing of mine...